packages/ember-cli-eyeglass/src/broccoli-ln-s.ts

```diff
@@ -0,0 +1,82 @@
+import BroccoliPlugin = require("broccoli-plugin");
+import path = require("path");
+import * as fs from "fs-extra";
+
+type EnsureSymlinkSync = (srcFile: string, destLink: string) => void;
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const ensureSymlink: EnsureSymlinkSync = require("ensure-symlink");
+
+/**
+ * An object where the keys are the files that will be created in the tree and
+ * the values are the source files.
+ * 
+ * @interface FileMap
+ */
+interface FileMap {
+  [relativePath: string]: string;
+}
+
+type BroccoliSymbolicLinkerOptions =
+  Pick<BroccoliPlugin.BroccoliPluginOptions, "annotation" | "persistentOutput">;
+
+/**
+ * Creates symlinks to the source files specified.
+ * 
+ *  BroccoliSymbolicLinker
+ */
+export class BroccoliSymbolicLinker extends BroccoliPlugin {
+  files: FileMap;
+  constructor(fileMap?: FileMap | undefined, options: BroccoliSymbolicLinkerOptions = {}) {
+    let pluginOpts: BroccoliPlugin.BroccoliPluginOptions = {needsCache: false};
+    Object.assign(pluginOpts, options);
+    super([], pluginOpts);
+    this.files = Object.assign({}, fileMap);
+  }
+  reset(fileMap?: FileMap | undefined): void {
+    this.files = Object.assign({}, fileMap);
+  }
+  /**
+   * Record that a symlink should be created from src to dest.
+   *
+   * This can be called many times before the build method is invoked.
+   * Calling it after will not have an effect until the next time build() is
+   * invoked.
+   *
+   * @param src The file that should be symlinked into the tree.
+   * @param dest the relative path from the tree's root to the location of the
+   *   symlink. the filename does not have to be the same.
+   * @returns the absolute path to the location where the symlink will be created.
+   */
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  ln_s(src: string, dest: string): string {
+    this.files[dest] = src;
```

Comment:
do we care if the `ln_s` is invoke twice with the same `dest` but differing `src` ? Is the rule simply last write wins?

Reply:
It would be a breaking change to eyeglass to do anything other than maintain the behavior of having the last write win.